distributed/worker.py

```diff
@@ -1683,7 +1683,7 @@ async def batched_send_connect():
 
                 bcomm.start(comm)
 
-            self.loop.add_callback(batched_send_connect)
```

Comment:
I think there are a few other instances of `loop.add_callback` you didn't refactor in worker.py, was that intentional?

Reply:
Some instances of this didn't handle cancellation in the prescribed order well. This has been left for future work.

Reply:
Maybe would be good to add TODO comments or open issues for those? It's nice when reading the code to be aware that something is an antipattern (and why)